### PR TITLE
imap-backup 4.0.2 (new formula)

### PR DESCRIPTION
**Pull request by jmechnich**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by joeyates** [comment]
Hi @jmechnich 

I'm the maintainer of the imap-backup gem. I have looked at the PR (joeyates/imap-backup#106).

Could this formula be implemented in a different way, by installing the [Ruby gem](https://rubygems.org/gems/imap-backup/) directly, instead of building the gem locally?

**Comment by jmechnich** [comment]
> Could this formula be implemented in a different way, by installing the [Ruby gem](https://rubygems.org/gems/imap-backup/) directly, instead of building the gem locally?

@SMillerDev @carlocab it seems that formulas exist using both approaches, is there a preferred way of doing it from your point if view?

**Comment by carlocab** [comment]
We should always build gems locally. See https://docs.brew.sh/Acceptable-Formulae#we-dont-like-binary-formulae

If there are formulae that don't do that and install binary objects, those should be fixed.

**Comment by carlocab** [comment]
```
==> Testing imap-backup
Error: imap-backup: failed
An exception occurred within a child process:
  Minitest::Assertion: Expected /^Choose an action:/ to match "exit without saving changes\r\n".
```

**Comment by jmechnich** [comment]
@carlocab yeah, I am a bit confused why it seems to work on arm (and now on ubuntu too), but fails on x86 macos. Unfortunately, I am on 12-arm64 myself for local testing but maybe I can figure something out...

**Comment by carlocab** [comment]
It's probably not really an ARM-Intel difference as it is a difference of running on a VM and running on bare metal. (Our Intel runners are VMs, the ARM runners are not.)

**Comment by jmechnich** [comment]
Something like the solution here might also be an option for working around the issue with EIO:
https://github.com/Homebrew/homebrew-core/blob/cdf1c999f81816eada6a2a0f503d405b67c47a5b/Formula/pianobar.rb#L41-L52

If this is preferred, I am happy to revert to the previous test using `read` instead of `readline`.

Also, the PR wrt to the `gemspec` file has just been accepted, so I am going to wait for the tag/release and make a final update to this PR.

File: Formula/imap-backup.rb
Comment (jmechnich) [reply Formula/imap-backup.rb]: @SMillerDev This should be resolved now (I hope)

File: Formula/imap-backup.rb
Comment (jmechnich) [reply Formula/imap-backup.rb]: These lines of code have been removed as the upstream PR has been merged.

**Comment by jmechnich** [comment]
@carlocab @SMillerDev Is there anything left you want me to improve or fix for this formula? If wanted, I can upgrade it to the most recent tag of imap-backup 4.0.2. The issues addressed in the comments should be resolved.

**Comment by SMillerDev** [comment]
If there is a newer version we should use that.

**Review by SMillerDev — APPROVED** [review]
Thanks @jmechnich  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**
